homeassistant/components/cover/__init__.py

```diff
@@ -0,0 +1,233 @@
+    return hass.states.is_state(entity_id, STATE_CLOSED)
+
+
+def open_cover(hass, entity_id=None):
+    """Open all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_OPEN_COVER, data)
+
+
+def close_cover(hass, entity_id=None):
+    """Close all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_CLOSE_COVER, data)
+
+
+def set_cover_position(hass, position, entity_id=None):
+    """Move to specific position all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    data[ATTR_POSITION] = position
+    hass.services.call(DOMAIN, SERVICE_SET_COVER_POSITION, data)
+
+
+def stop_cover(hass, entity_id=None):
+    """Stop all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_STOP_COVER, data)
+
+
+def open_cover_tilt(hass, entity_id=None):
+    """Open all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_OPEN_COVER_TILT, data)
+
+
+def close_cover_tilt(hass, entity_id=None):
+    """Close all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_CLOSE_COVER_TILT, data)
+
+
+def set_cover_tilt_position(hass, tilt_position, entity_id=None):
+    """Move to specific tilt position all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    data[ATTR_TILT_POSITION] = tilt_position
+    hass.services.call(DOMAIN, SERVICE_SET_COVER_TILT_POSITION, data)
+
+
+def stop_cover_tilt(hass, entity_id=None):
+    """Stop all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_STOP_COVER_TILT, data)
+
+
+def setup(hass, config):
+    """Track states and offer events for covers."""
+    component = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL, GROUP_NAME_ALL_COVERS)
+    component.setup(config)
+
+    def handle_cover_service(service):
+        """Handle calls to the cover services."""
+        method = SERVICE_TO_METHOD.get(service.service)
+        params = service.data.copy()
+        params.pop(ATTR_ENTITY_ID, None)
+
+        if method:
+            for cover in component.extract_from_service(service):
+                getattr(cover, method['method'])(**params)
+
+                if cover.should_poll:
+                    cover.update_ha_state(True)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    service_name = None
+    for service_name in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[service_name].get(
+            'schema', COVER_SERVICE_SCHEMA)
+    hass.services.register(DOMAIN, service_name, handle_cover_service,
```

Comment:
This should be indented to run within the for loop. You don't need to initialize `service_name` to `None` then either.